swap_meet/vendor.py

```diff
@@ -1,2 +1,106 @@
+from .item import Item
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory= None): 
+        self.inventory = inventory if inventory is not None else []
+    
+# Methods 
+
+    def add(self, item):
+        ''' adds item to inventory '''
+        self.inventory.append(item)
+        return item 
+
+
+    def remove(self, other_item):
+        ''' removes duplicate item if true '''
+        if other_item in self.inventory:
+            self.inventory.remove(other_item)
+            return other_item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        ''' returns a list of items in the inventory
+            if item category matches item category in inventory
+        '''
+        list_of_items= []
+        for item in self.inventory:
+            if item.category == category:
+                list_of_items.append(item)
+        return list_of_items
+
+
+    def swap_items(self, vendor_friend, my_item, their_item):
+        '''
+        return False if inventory doesn't contain my_item or 
+        vendor friend doesn't contain their_item
+
+        return True 
+        '''
+        if my_item not in self.inventory or their_item not in vendor_friend.inventory:
+            return False
+        else:
+            self.remove(my_item)
+            vendor_friend.add(my_item)
+            vendor_friend.remove(their_item)
+            self.add(their_item)
+            return True
+
+    def swap_first_item(self, swapping_friend):
+        '''
+        if self.inventory or swapping_friend.inventory
+        are empty, return False
+
+        remove 1st item from self.inventory add swapping_friend 1st item
+        remove ""       ""   swapping_friend.inventory add self.inventory ""
+        return True
+        '''
+
+            
+        if not self.inventory or not swapping_friend.inventory:
+            return False
+        else:
+            self.add(swapping_friend.inventory[0])
+            swapping_friend.add(self.inventory[0])
+            self.remove(self.inventory[0])
+            swapping_friend.remove(swapping_friend.inventory[0])
+            return True
```

Comment:
Hmm have we seen this before? Could we use `swap_items` instead so we aren't duplicating code?